packages/api/src/command/medical/cq-directory/parse-cq-directory-entry.ts

```diff
@@ -0,0 +1,99 @@
+import { XCA_DQ_STRING, XCA_DR_STRING, XCPD_STRING } from "@metriport/carequality-sdk/common/util";
+import { Address } from "@metriport/carequality-sdk/models/address";
+import { Contained } from "@metriport/carequality-sdk/models/contained";
+import { Organization } from "@metriport/carequality-sdk/models/organization";
+import { normalizeOid } from "@metriport/shared";
+import { CQDirectoryEntryData } from "../../../domain/medical/cq-directory";
+
+export type XCUrls = {
+  urlXCPD: string;
+  urlDQ?: string;
+  urlDR?: string;
+};
+
+export function parseCQDirectoryEntries(orgsInput: Organization[]): CQDirectoryEntryData[] {
+  const orgs = orgsInput.flatMap(org => {
+    const orgOid = org?.identifier?.value?.value;
+    if (!orgOid) return [];
+
+    const url = getUrls(org?.contained);
+    if (!url?.urlXCPD) return [];
+    let oid = org?.identifier?.value?.value;
+    try {
+      oid = normalizeOid(org?.identifier?.value?.value);
+    } catch (err) {
+      console.log(`Organization ${org.name?.value} has invalid OID: ${oid}`);
+      return [];
+    }
```

Comment:
Not a huge deal, but its confusing to init a variable to then always update with something else; we could just create `oid: string | undefined = undefined;`